modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -604,4 +608,200 @@ private void deleteApplication(String applicationId) throws Exception {
+        Assert.assertEquals(actualOrganizationCount, expectedOrganizationCount);
+
+        String nextLink = response.jsonPath().getString("links.find { it.rel == 'next' }.href");
+        String afterValue = null;
+
+        if (nextLink != null && nextLink.contains(AFTER_CURSOR_ARG)) {
+            afterValue = nextLink.substring(nextLink.indexOf(AFTER_CURSOR_ARG) + 6);
+        }
+
+        String storedAfterValue = afterValue;
+
+        if (totalCreatedOrganizations > limit) {
+            Assert.assertNotNull(storedAfterValue);
+        } else {
+            Assert.assertNull(storedAfterValue);
+        }
+    }
+
+    @Test(dependsOnMethods = "createOrganizationsForPaginationTests")
+    public void testGetPaginatedOrganizationsWithCursor() {
+
+        String after;
+        String before;
+        int limit = 2;
+        int orgLimit = 20;
+        int largeLimit = 30;
+
+        // Step 1: Call the first page
+        String firstPageUrl = ORGANIZATION_MANAGEMENT_API_BASE_PATH + LIMIT_ARG + limit + "&recursive=false";
+        Response firstPageResponse = getResponseOfGetWithOAuth2(firstPageUrl, m2mToken);
+
+        validateHttpStatusCode(firstPageResponse, HttpStatus.SC_OK);
+
+        int firstPageOrgCount = firstPageResponse.jsonPath().getList("organizations").size();
+        Assert.assertEquals(firstPageOrgCount, Math.min(limit, totalCreatedOrganizations));
+
+        List<Map<String, String>> firstPageLinks = firstPageResponse.jsonPath().getList("links");
+        after = getLink(firstPageLinks, "next");
+        before = getLink(firstPageLinks, "previous");
+
+        Assert.assertNotNull(after, "After value should not be null on the first page.");
+        Assert.assertNull(before, "Before value should be null on the first page.");
+
+        // Step 2: Call the second page using the 'after' value
+        String secondPageUrl = ORGANIZATION_MANAGEMENT_API_BASE_PATH + LIMIT_ARG + limit
+                + "&recursive=false&after=" + after;
+        Response secondPageResponse = getResponseOfGetWithOAuth2(secondPageUrl, m2mToken);
+
+        validateHttpStatusCode(secondPageResponse, HttpStatus.SC_OK);
+
+        int secondPageOrgCount = secondPageResponse.jsonPath().getList("organizations").size();
+        Assert.assertEquals(secondPageOrgCount, Math.min(limit, totalCreatedOrganizations - firstPageOrgCount));
+
+        List<Map<String, String>> secondPageLinks = secondPageResponse.jsonPath().getList("links");
+        before = getLink(secondPageLinks, "previous");
+        after = getLink(secondPageLinks, "next");
+
+        Assert.assertNotNull(before, "Before value should not be null on the second page.");
+        if (totalCreatedOrganizations > limit * 2) {
+            Assert.assertNotNull(after, "After value should not be null if there are more pages.");
+        } else {
+            Assert.assertNull(after, "After value should be null if this is the last page.");
+        }
```

Comment:
Here, do we check if the organizations list returned on the second page is the correct list of organizations? Otherwise, an issue like https://github.com/wso2/product-is/issues/20700 could occur, where the returned list differs from the expected list.

Reply:
Done